modules/spring-config/src/main/java/org/mule/runtime/config/internal/el/DataWeaveExtendedExpressionLanguageAdaptorFactoryBean.java

```diff
@@ -15,9 +17,15 @@
 
 import javax.inject.Inject;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.DisposableBean;
 import org.springframework.beans.factory.FactoryBean;
 
-public class DataWeaveExtendedExpressionLanguageAdaptorFactoryBean implements FactoryBean<ExtendedExpressionLanguageAdaptor> {
+public class DataWeaveExtendedExpressionLanguageAdaptorFactoryBean implements FactoryBean<ExtendedExpressionLanguageAdaptor>,
```

Comment:
This solution relies heavily in assuming that a Spring registry which is not the case in Mule Framework. I think that the actual issue is that the DataWeaveExpressionLanguageAdaptor class does implement `Disposable` and lifecycle should be applied to it regardless of how it got there. Do you have more details of how and why this happens? The GUS issue doesn't provide many details and if this also happens in MuleFramework this solution won't cut it.

Reply:
FactoryBean is a spring class (so in that sense at this level it is assuming it is a Spring Registry otherwise) and DataWeaveExpressionLanguageAdaptor is Disposable.

Reply:
The gus ticket has a parent which is related to munit failing tests.